MaxText/layers/linears.py

```diff
@@ -105,6 +106,9 @@ def compute_dot_general(inputs, kernel, axis, contract_ind):
       if self.quant:
         dot_general_cls = self.quant.dot_general_cls()
         dot_general = dot_general_cls()
+      if self.use_fp8:
```

Comment:
Functionally, this looks great!

But, I think it may be cleaner if we added some conditional logic here instead `https://github.com/google/maxtext/blob/main/MaxText/layers/quantizations.py#L95`. We could introduce a general Quantization class (and existing AqtQuantization could subclass). We could then have an FP8 implementation of that class that returns nn.Fp8DotGeneralOp when dot_general_cls() is called. That is a little bit more work but feels a bit better to me. But open to hear other thoughts.

Reply:
Made some changes. I didn't know what the general Quantization class should look like. So I've omitted it for now. Lmk if you'd still like me to introduce it.

Reply:
Awesome! Thanks for this. I've left a comment about what I think it could look like but let me know your thoughts!